python/activator/middleware_interface.py

```diff
@@ -56,25 +58,46 @@ class MiddlewareInterface:
         Path to a butler repo containing the calibration and other data needed
         for processing images as they are received.
     image_bucket : `str`
-        Google storage bucket where images will be written to as they arrive.
+        Storage bucket where images will be written to as they arrive.
+        See also ``prefix``.
     instrument : `str`
-        Name of the instrument taking the data, for populating butler
-        collections and dataIds.
+        Full class name of the instrument taking the data, for populating
+        butler collections and dataIds. Example: "lsst.obs.lsst.LsstCam"
```

Comment:
If you want to keep the interface in terms of the full class name, activator.py could do the lookup from short name to full class name.

Reply:
Butler knows how to convert butler instrument name to class name. See also `Instrument.fromName()`.

Reply:
I'm not currently focusing on keeping activator and MiddlewareInterface in sync at present, just trying to make the MiddlewareInterface API make the most sense I can. I think we'll probably have to live with some churn, as we flesh out each part of the API (MI, activator, uploading), since exactly how they're written right now may not be the final thing we want.

Given @timj 's suggestion, I don't know whether using the full class name or a short name is actually more appropriate here. The former is much more explicit, but the latter is better for log message prefixes. What are the arguments for using one or the other?

Reply:
I just checked, and `Instrument.fromName` requires that the instrument be registered. So you need the long name to do the registration (modulo my later comment about `import_` taking care of this for you).

I strongly oppose letting `activator` bit-rot while you try to perfect `middleware`. Not only does it increase the chance that you'll spend a lot of time on decisions that will later turn out to have not made sense, but it makes it impossible to develop the Prompt Processing service incrementally and opens up the risk that we will be forced to leave the project in an unusable state for the next person/group to work on it.

Reply:
But we cannot use `activator` at all if `MiddlewareInterface` has code in it involving actual files. As it stands, even if nothing broke at the `activator`/`middleware` interface on this PR, trying to use it in gcloud would fail because there is no test data to run it on. By implementing this code without having test data in place (and test "upload" code that uses that test data), we've already left it in an "unusable" state.

Reply:
But we _will_ have test data long before `MiddlewareInterface` is finalized.

Reply:
@kfindeisen and I talked this through, and I've attempted to update activator.py to have it follow `middleware` changes, or at least mention where things may have diverged.